packages/datetime/src/common/dateUtils.ts

```diff
@@ -4,27 +4,26 @@
  * Licensed under the terms of the LICENSE file distributed with this project.
  */
 
-import * as moment from "moment";
+import * as format from "date-fns/format";
+import * as isSameDay from "date-fns/is_same_day";
+import * as parse from "date-fns/parse";
+
+// Import a subset of locales to reduce the bundle size of consuming apps
+// Following locales have been selected according to their popularity
+const LOCALE: { [index: string]: any } = {
+    de: require("date-fns/locale/de"),
```

Comment:
these are static requires. currently, the locales get imported as soon as `@blueprintjs/datetime` is imported, which isn't what we want. users should be able to supply their own list of locales from `date-fns` if we end up going this route.

Reply:
These static requires are intended to be a workaround until `date-fns` releases a `v2.0` in which it's going to be possible to support user specified locales.

There also are plans for `date-fns` to support timezones:
 - https://github.com/date-fns/date-fns/issues/180
 - https://github.com/date-fns/date-fns/issues/376

I think date-fns have good potential, but it needs to improve and is under active development.

Another lib that might be worth exploring : https://github.com/smallwins/spacetime